tests/system/src/test/java/test/robot/javafx/stage/AttributesTest.java

```diff
@@ -0,0 +1,187 @@
+        final CountDownLatch topShownLatch = new CountDownLatch(1);
+
+        runAndWait(() -> {
+            // Bottom stage, should be visible after top stage is iconified
+            bottomStage = getStage(false);
+            bottomStage.initStyle(StageStyle.DECORATED);
+            Scene bottomScene = new Scene(new Pane(), WIDTH, HEIGHT);
+            bottomScene.setFill(BOTTOM_COLOR);
+            bottomStage.setScene(bottomScene);
+            bottomStage.setX(0);
+            bottomStage.setY(0);
+            bottomStage.setOnShown(e -> Platform.runLater(bottomShownLatch::countDown));
+            bottomStage.show();
+        });
+
+        assertTrue("Timeout waiting for bottom stage to be shown",
+            bottomShownLatch.await(TIMEOUT, TimeUnit.MILLISECONDS));
+
+        runAndWait(() -> {
+            // Top stage, will be inconified
+            topStage = getStage(true);
+            topStage.initStyle(StageStyle.DECORATED);
+            Scene topScene = new Scene(new Pane(), WIDTH, HEIGHT);
+            topScene.setFill(TOP_COLOR);
+            topStage.setScene(topScene);
+            if (overlayed) {
+                topStage.setX(0);
+                topStage.setY(0);
+            } else {
+                topStage.setX(WIDTH);
+                topStage.setY(HEIGHT);
+            }
+            topStage.setOnShown(e -> Platform.runLater(topShownLatch::countDown));
+            topStage.show();
+        });
+
+        assertTrue("Timeout waiting for top stage to be shown",
+            topShownLatch.await(TIMEOUT, TimeUnit.MILLISECONDS));
+    }
+
+    public @Test void testIconifiedStage() throws InterruptedException {
+        setupStages(true);
+
+        runAndWait(() -> {
+            Color color = getColor(200, 200);
+            assertColorEquals(TOP_COLOR, color, TOLERANCE);
+
+            topStage.setIconified(true);
+        });
+
+        // wait a bit to let window system animate the change
+        sleep(1000);
+
+        runAndWait(() -> {
+            assertTrue(topStage.isIconified());
+            Color color = getColor(200, 200);
+            assertColorEquals(BOTTOM_COLOR, color, TOLERANCE);
+        });
+    }
+
+    public @Test void testMaximizedStage() throws InterruptedException {
+        setupStages(false);
+
+        runAndWait(() -> {
+            Color color = getColor(200, 200);
+            assertColorEquals(BOTTOM_COLOR, color, TOLERANCE);
+
+            topStage.setMaximized(true);
+        });
+
+        // wait a bit to let window system animate the change
+        sleep(1000);
+
+        runAndWait(() -> {
+            assertTrue(topStage.isMaximized());
+
+            // maximized stage should take over the bottom stage
+            Color color = getColor(200, 200);
+            assertColorEquals(TOP_COLOR, color, TOLERANCE);
+        });
+
+        // wait a little bit between getColor() calls - on macOS the below one
+        // would fail without this wait
+        sleep(100);
```

Comment:
same - waitForIdle?

Reply:
I don't think so. This is a small sleep that isn't intended to wait for idle.